File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels=['a','e','i','o','u']
        a=s[:len(s)//2]
        b=s[len(s)//2:]
        vowels_in_a=0
        vowels_in_b=0

        for c in a:
            if c.lower() in vowels:
                vowels_in_a+=1
        for c in b:
            if c.lower() in vowels:
                vowels_in_b+=1

        return vowels_in_a==vowels_in_b
        

        

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.js
/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function(s) {
    const vowels=["a","e","i","o","u"]
    const a=s.split("").slice(0,s.length/2)
    const b=s.split("").slice(s.length/2)
    let vowelCountOfA=0
    let vowelCountOfB=0

    for(let i=0;i<a.length;i++){
        if(vowels.includes(a[i].toLowerCase())){
            vowelCountOfA++
        }
    }
    for(let i=0;i<b.length;i++){
        if(vowels.includes(b[i].toLowerCase())){
            vowelCountOfB++
        }
    }
    
    return vowelCountOfA===vowelCountOfB
        
};